### ISSUE TITLE
Bug trace thread

### ISSUE THREAD

**Issue by EclectickMedia** [EclectickMedia/PingStats#39]
From #37:

```
Mon Dec  5 21:47:17 2016
os.name: nt
platform.system: Windows
platform.release: 7
PingStats version: 2.0.02
Python version (via sys.version): 3.4.3 (v3.4.3:9b73f1c3e601, Feb 24 2015, 22:44
:40) [MSC v.1600 64 bit (AMD64)]
stdout:
b'3e75937 Forced buffered output of unittests\n'
stderr:
b''


.F
Stdout:
Falsifying example: test_instantiate_catch_null_byte_title_string(self=<__main__
.BasePlot_test testMethod=test_instantiate_catch_null_byte_title_string>, data='
\x00')
............E
Stdout:
Falsifying example: test_build_files(self=<__main__.TestCore testMethod=test_bui
ld_files>, path='.', name='*')
E
Stdout:
Request timed out.
Request timed out.
Request timed out.
Request timed out.
Request timed out.
Request timed out.
Request timed out.
Request timed out.
Request timed out.
Falsifying example: test_failed_connections(self=<__main__.TestCore testMethod=t
est_failed_connections>, address='\x85')

Stderr:
C:\Python34\lib\site-packages\hypothesis\internal\conjecture\engine.py:86: Resou
rceWarning: unclosed <socket.socket fd=736, family=AddressFamily.AF_INET, type=S
ocketKind.SOCK_RAW, proto=1>
  raise e
C:\Python34\lib\unittest\case.py:605: ResourceWarning: unclosed <socket.socket f
d=736, family=AddressFamily.AF_INET, type=SocketKind.SOCK_RAW, proto=1>
  outcome.errors.clear()
.
======================================================================
ERROR: test_build_files (__main__.TestCore)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests.py", line 21, in test_build_files
    def test_build_files(self, path, name):
  File "C:\Python34\lib\site-packages\hypothesis\core.py", line 524, in wrapped_
test
    print_example=True, is_final=True
  File "C:\Python34\lib\site-packages\hypothesis\executors.py", line 58, in defa
ult_new_style_executor
    return function(data)
  File "C:\Python34\lib\site-packages\hypothesis\core.py", line 111, in run
    return test(*args, **kwargs)
  File "tests.py", line 23, in test_build_files
    test_file = PingStats.buildfile(path, name)
  File "A:\Maya\Projects\PingStats\PingStats.py", line 69, in buildfile
    return open(path + name, 'a+')
OSError: [Errno 22] Invalid argument: '.*.csv'

Stdout:
Falsifying example: test_build_files(self=<__main__.TestCore testMethod=test_bui
ld_files>, path='.', name='*')

======================================================================
ERROR: test_failed_connections (__main__.TestCore)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Python34\lib\encodings\idna.py", line 181, in encode
    result.extend(ToASCII(label))
  File "C:\Python34\lib\encodings\idna.py", line 76, in ToASCII
    label = nameprep(label)
  File "C:\Python34\lib\encodings\idna.py", line 38, in nameprep
    raise UnicodeError("Invalid character %r" % c)
UnicodeError: Invalid character '\x85'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "tests.py", line 52, in test_failed_connections
    def test_failed_connections(self, address):
  File "C:\Python34\lib\site-packages\hypothesis\core.py", line 524, in wrapped_
test
    print_example=True, is_final=True
  File "C:\Python34\lib\site-packages\hypothesis\executors.py", line 58, in defa
ult_new_style_executor
    return function(data)
  File "C:\Python34\lib\site-packages\hypothesis\core.py", line 111, in run
    return test(*args, **kwargs)
  File "tests.py", line 53, in test_failed_connections
    self.assertEqual(PingStats.ping(address, verbose=False).__next__()[1],
  File "A:\Maya\Projects\PingStats\PingStats.py", line 92, in ping
    verbose=verbose)[0],
  File "A:\Maya\Projects\PingStats\pythonping\ping.py", line 140, in single_ping

    verbose)
  File "A:\Maya\Projects\PingStats\pythonping\ping.py", line 241, in _send
    mySocket.sendto(packet, (destIP, 1))  # Port number is irrelevant
UnicodeError: encoding with 'idna' codec failed (UnicodeError: Invalid character
 '\x85')

Stdout:
Request timed out.
Request timed out.
Request timed out.
Request timed out.
Request timed out.
Request timed out.
Request timed out.
Request timed out.
Request timed out.
Falsifying example: test_failed_connections(self=<__main__.TestCore testMethod=t
est_failed_connections>, address='\x85')

Stderr:
C:\Python34\lib\site-packages\hypothesis\internal\conjecture\engine.py:86: Resou
rceWarning: unclosed <socket.socket fd=736, family=AddressFamily.AF_INET, type=S
ocketKind.SOCK_RAW, proto=1>
  raise e

======================================================================
FAIL: test_instantiate_catch_null_byte_title_string (__main__.BasePlot_test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests.py", line 130, in test_instantiate_catch_null_byte_title_string
    def test_instantiate_catch_null_byte_title_string(self, data):
  File "C:\Python34\lib\site-packages\hypothesis\core.py", line 524, in wrapped_
test
    print_example=True, is_final=True
  File "C:\Python34\lib\site-packages\hypothesis\executors.py", line 58, in defa
ult_new_style_executor
    return function(data)
  File "C:\Python34\lib\site-packages\hypothesis\core.py", line 111, in run
    return test(*args, **kwargs)
  File "tests.py", line 134, in test_instantiate_catch_null_byte_title_string
    plot()
AssertionError: TypeError not raised

Stdout:
Falsifying example: test_instantiate_catch_null_byte_title_string(self=<__main__
.BasePlot_test testMethod=test_instantiate_catch_null_byte_title_string>, data='
\x00')

----------------------------------------------------------------------
Ran 17 tests in 64.506s

FAILED (failures=1, errors=2)
```

**Comment by EclectickMedia** [comment]
Commit 9332a4d999abafcb2bb0e814c09d10df687e8f4c should clear out a few issues on this debug. 

Run it again and reattach?

**Comment by frawst** [comment]
```
Wed Dec  7 04:44:41 2016
os.name: nt
platform.system: Windows
platform.release: 7
PingStats version: 2.0.02
Python version (via sys.version): 3.4.3 (v3.4.3:9b73f1c3e601, Feb 24 2015, 22:44
:40) [MSC v.1600 64 bit (AMD64)]
stdout:
b'73f7829 Updated write_csv_data docs\n'
stderr:
b''


.F
Stdout:
Falsifying example: test_instantiate_catch_null_byte_title_string(self=<__main__
.BasePlot_test testMethod=test_instantiate_catch_null_byte_title_string>, data='
\x00')
............E
Stdout:
Falsifying example: test_build_files(self=<__main__.TestCore testMethod=test_bui
ld_files>, path='.', name='*')
...
======================================================================
ERROR: test_build_files (__main__.TestCore)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests.py", line 21, in test_build_files
    def test_build_files(self, path, name):
  File "C:\Python34\lib\site-packages\hypothesis\core.py", line 524, in wrapped_
test
    print_example=True, is_final=True
  File "C:\Python34\lib\site-packages\hypothesis\executors.py", line 58, in defa
ult_new_style_executor
    return function(data)
  File "C:\Python34\lib\site-packages\hypothesis\core.py", line 111, in run
    return test(*args, **kwargs)
  File "tests.py", line 23, in test_build_files
    test_file = PingStats.buildfile(path, name)
  File "A:\Maya\Projects\PingStats\PingStats.py", line 68, in buildfile
    raise ValueError('Illegal file name %s' % name)
ValueError: Illegal file name *

Stdout:
Falsifying example: test_build_files(self=<__main__.TestCore testMethod=test_bui
ld_files>, path='.', name='*')

======================================================================
FAIL: test_instantiate_catch_null_byte_title_string (__main__.BasePlot_test)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests.py", line 135, in test_instantiate_catch_null_byte_title_string
    def test_instantiate_catch_null_byte_title_string(self, data):
  File "C:\Python34\lib\site-packages\hypothesis\core.py", line 524, in wrapped_
test
    print_example=True, is_final=True
  File "C:\Python34\lib\site-packages\hypothesis\executors.py", line 58, in defa
ult_new_style_executor
    return function(data)
  File "C:\Python34\lib\site-packages\hypothesis\core.py", line 111, in run
    return test(*args, **kwargs)
  File "tests.py", line 139, in test_instantiate_catch_null_byte_title_string
    plot()
AssertionError: TypeError not raised

Stdout:
Falsifying example: test_instantiate_catch_null_byte_title_string(self=<__main__
.BasePlot_test testMethod=test_instantiate_catch_null_byte_title_string>, data='
\x00')

----------------------------------------------------------------------
Ran 18 tests in 2.607s

FAILED (failures=1, errors=1)
```

**Comment by EclectickMedia** [comment]
Commit 2973f3c171dc43bfd41a22999adf7e0c222e443e should fix all errors in this debug.

**Comment by frawst** [comment]
commit https://github.com/EclectickMedia/PingStats/commit/2973f3c171dc43bfd41a22999adf7e0c222e443e : 

```
Wed Dec  7 19:48:54 2016
os.name: nt
platform.system: Windows
platform.release: 7
PingStats version: 2.0.02
Python version (via sys.version): 3.4.3 (v3.4.3:9b73f1c3e601, Feb 24 2015, 22:44
:40) [MSC v.1600 64 bit (AMD64)]
stdout:
b'2973f3c Added logic to fail if the wrong exception is caught for null byte tit
le strings\n'
stderr:
b''


..............E
Stdout:
Caught illegal file name exception
Caught illegal file name exception
Falsifying example: test_build_files(self=<__main__.TestCore testMethod=test_bui
ld_files>, path='.', name='"')
...
======================================================================
ERROR: test_build_files (__main__.TestCore)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests.py", line 21, in test_build_files
    def test_build_files(self, path, name):
  File "C:\Python34\lib\site-packages\hypothesis\core.py", line 524, in wrapped_
test
    print_example=True, is_final=True
  File "C:\Python34\lib\site-packages\hypothesis\executors.py", line 58, in defa
ult_new_style_executor
    return function(data)
  File "C:\Python34\lib\site-packages\hypothesis\core.py", line 111, in run
    return test(*args, **kwargs)
  File "tests.py", line 23, in test_build_files
    test_file = PingStats.buildfile(path, name)
  File "A:\Maya\Projects\PingStats\PingStats.py", line 73, in buildfile
    return open(path + name, 'a+')
OSError: [Errno 22] Invalid argument: '.".csv'

Stdout:
Caught illegal file name exception
Caught illegal file name exception
Falsifying example: test_build_files(self=<__main__.TestCore testMethod=test_bui
ld_files>, path='.', name='"')

----------------------------------------------------------------------
Ran 18 tests in 3.637s

FAILED (errors=1)
```

Commit https://github.com/EclectickMedia/PingStats/commit/ad4a11063fa8a1b2f6c7d1c174b088f31a5c4856 : 

```
Wed Dec  7 19:50:15 2016
os.name: nt
platform.system: Windows
platform.release: 7
PingStats version: 2.0.02
Python version (via sys.version): 3.4.3 (v3.4.3:9b73f1c3e601, Feb 24 2015, 22:44
:40) [MSC v.1600 64 bit (AMD64)]
stdout:
b'ad4a110 removed data validation from PlotTable, updated tests.\n'
stderr:
b''


.....s..s.....E
Stdout:
Caught illegal file name exception
Caught illegal file name exception
Falsifying example: test_build_files(self=<__main__.TestCore testMethod=test_bui
ld_files>, path='.', name='"')
...
======================================================================
ERROR: test_build_files (__main__.TestCore)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests.py", line 21, in test_build_files
    def test_build_files(self, path, name):
  File "C:\Python34\lib\site-packages\hypothesis\core.py", line 524, in wrapped_
test
    print_example=True, is_final=True
  File "C:\Python34\lib\site-packages\hypothesis\executors.py", line 58, in defa
ult_new_style_executor
    return function(data)
  File "C:\Python34\lib\site-packages\hypothesis\core.py", line 111, in run
    return test(*args, **kwargs)
  File "tests.py", line 23, in test_build_files
    test_file = PingStats.buildfile(path, name)
  File "A:\Maya\Projects\PingStats\PingStats.py", line 56, in buildfile
    return open(path + name, 'a+')
OSError: [Errno 22] Invalid argument: '.".csv'

Stdout:
Caught illegal file name exception
Caught illegal file name exception
Falsifying example: test_build_files(self=<__main__.TestCore testMethod=test_bui
ld_files>, path='.', name='"')

----------------------------------------------------------------------
Ran 18 tests in 3.129s

FAILED (errors=1, skipped=2)
```

**Comment by EclectickMedia** [comment]
Thanks. Theres a new commit up that changes the import structure a bit, bit I need to add a clause for the new name error...

This isnt to serious of a bug, just dont name your files `"`...

**Comment by EclectickMedia** [comment]
Closed by ecb44594aca563401f935043396324d38c2158b1
